core/types/transaction.go

```diff
@@ -200,14 +200,17 @@ func (tx *Transaction) decodeTyped(b []byte) (TxData, error) {
 	if len(b) <= 1 {
 		return nil, errShortTypedTx
 	}
+
+	if inner, isCelo, err := celoDecodeTyped(b); isCelo {
+		return inner, err
+	}
+
```

Comment:
This looks unnecessarily abstract in this diff, but with more than one Celo tx type it becomes worth it.

Reply:
When I added this I was envisaging having at least 2 celo types, however even with just one type I think it could provide an advantage since I imagine there would be more activity in the switch block (every time a tx is added or removed) as opposed to outside of the switch block. So less likely to see conflicts there.